File: backend/src/scripts/vai_xanh_la.py
import json
import os
import sys

# Add the backend/src directory to the Python path
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from libs.adb_controller import AdbController
from models.game_options import GameOptions
from scripts.core import (
    SellOption,
    go_down,
    go_last,
    go_up,
    harvest_tree,
    make_items,
    open_chest,
    open_game,
    plant_tree,
    sell_items,
)


def main():
    if len(sys.argv) < 2:
        print("Usage: python vai_xanh_la.py <device_id> [game_options_json]")
        sys.exit(1)

    device_id = sys.argv[1]
    game_options = GameOptions()

    # Parse game options if provided
    if len(sys.argv) > 2:
        try:
            options_dict = json.loads(sys.argv[2])
            game_options = GameOptions.from_dict(options_dict)
        except json.JSONDecodeError:
            print("Error: Invalid JSON format for game_options")
            sys.exit(1)

    manager = AdbController(device_id)

    if game_options.open_game:
        open_game(manager)

    for i in range(100):
        print(f"{i}: Run vai xanh la")
        if game_options.open_chest:
            open_chest(manager)

        for _ in range(10):
            # auto
            go_up(manager)
            plant_tree(manager, "chanh")
            go_up(manager, 2)
            plant_tree(manager, "chanh")
            go_down(manager, 2)
            manager.sleep(6)

            harvest_tree(manager)
            plant_tree(manager, "bong", next=False)
            go_up(manager, 2)
            harvest_tree(manager)
            plant_tree(manager, "bong", 6)
            go_down(manager, 2)

            make_items(manager, 2, 3, 6)
            harvest_tree(manager)
            go_up(manager, 2)
            harvest_tree(manager)
            make_items(manager, 1, 3, 6)
            go_last(manager)

        if game_options.sell_items:
            sell_items(
                manager,
                SellOption.GOODS,
                [{"key": "vai-xanh-la", "value": 6}],
            )

        i += 1

    print("Vai xanh la automation completed")


if __name__ == "__main__":
    main()

File: api/Automation/Scripts/VaiXanhLa.cs
using KvtmAuto.Core.Models;

namespace KvtmAuto.Automation.Scripts;

public class VaiXanhLa(AdbController adb, ImageMatcher images, DeviceManager deviceManager)
    : ScriptBase(adb, images, deviceManager)
{
    public override string Id => "02";
    public override string Name => "Vai Xanh La";

    public override async Task RunAsync(string deviceId, GameOptions options, CancellationToken ct)
    {
        if (options.OpenGame)
            await OpenGameAsync(deviceId, ct);

        for (int i = 0; i < LoopNum; i++)
        {
            ct.ThrowIfCancellationRequested();
            Log(deviceId, $"{i}: Run vai xanh la");

            if (options.OpenChests)
                await OpenChestAsync(deviceId, ct);

            for (int j = 0; j < 10; j++)
            {
                await GoUpAsync(deviceId, ct);
                await PlantTreeAsync(deviceId, ct, "chanh");
                await GoLastAsync(deviceId, ct);
                await SleepAsync(7, ct);

                await GoUpAsync(deviceId, ct);
                await HarvestTreeAsync(deviceId, ct);
                await PlantTreeAsync(deviceId, ct, "bong", num: 18, next: false);
                await GoLastAsync(deviceId, ct);

                await GoUpAsync(deviceId, ct);
                await MakeItemsAsync(deviceId, ct, floor: 2, slot: 3, num: 6);
                await HarvestTreeAsync(deviceId, ct);
                await GoUpAsync(deviceId, ct, 2);
                await MakeItemsAsync(deviceId, ct, floor: 1, slot: 3, num: 6);
                await GoLastAsync(deviceId, ct);
            }

            if (options.SellItems)
                await SellItemsAsync(deviceId, ct, SellOption.Goods,
                [
                    new SellItem("vai-xanh-la", 6),
                ]);

            await MakeEventAsync(deviceId, ct);
        }

        Log(deviceId, "The automation completed");
    }
}
